### ISSUE TITLE
QL ratings (tag) not respected on different machine

### ISSUE THREAD

**Issue by sebschlicht** [quodlibet/quodlibet#3859]
Steps to reproduce
------------------
1. Enable *Save ratings and play counts in tags* and enter the same Email for rating tags in QL on two machines A and B
   * value used for the `Email` field: `banshee`
2. Play and rate songs (Ogg Vorbis files) on machine A
3. Copy the files to machine B
4. Import the files to your library on machine B

Expected Output
---------------
QL should show the ratings that have been performed on machine A when playing the files on machine B.

Actual Output
-------------
QL shows the default rating for all those files.

Test System
-----------

### Which version of Quod Libet?
4.4.0 on both machines

### Which operating system 
Ubuntu 20.04 LTS on both machines

Additional Information
----------------------
I can see that the rating tag does exist, using `ffmpeg -i` on one of those files.
Here is an example extract from the output of this command on machine B

    ....
    Input #0, ogg, from '<title>.ogg':
      Duration: 00:02:53.24, start: 0.000000, bitrate: 425 kb/s
      Stream #0:0: Audio: vorbis, 44100 Hz, stereo, fltp, 320 kb/s
      Metadata:
      ...
      RATING:BANSHEE  : 0.4
      ...

but sometimes the tag is lower case instead

    ...
    rating:banshee  : 0.8
    ...

I noted that the expected output is shown for a single file in my library, where the Email value was `quodlibet@lists.sacredchao.net`, the respective `ffmpeg` output:

    ...
    rating:quodlibet@lists.sacredchao.net: 0.8
    ...

Might QL fail to read ratings in case the value was not a valid mail address?

**Comment by declension** [comment]
> Might QL fail to read ratings in case the Email value was not a valid mail address?
I've never tried, but I was thinking this is probably the cause too

**Comment by declension** [comment]
Did you ensure the ratings were saved before copying? This is trickier than it seems (use the _Update Tags in Files_ plugin to be sure)

**Comment by sebschlicht** [comment]
I can try to use this plugin and copy the files again, but I'd love to approach this more a more general direction too: How can I reliably check whether the tags exist?

The `ffmpeg` output apparently is not sufficient. I just checked `vorbiscomment` but it gives me the same results:

    // rating not recognized
    RATING:BANSHEE=0.8
    // rating of 0.8 recognized
    rating:quodlibet@lists.sacredchao.net=0.8
    rating:banshee=0.6

Which tool do you use to be sure here?

**Comment by declension** [comment]
Yes that should be fine (thought the finer details are probably more of an [mutagen](/quodlibet/mutagen) thing.
I can see similar results with `vorbiscomment` (just installed - I don't use ogg much). 

In the second file from what I can see, QL seems to be successfully persisting ratings for two users, `banshee` and `quodlibet@lists.sacredchao.net` (the default, so handled slightly differently in places)

I think the key here is that the ALL-CAPS isn't working, no? 

> Alright, whatever QL is using, it's not the tags shown by these tools: I tried to change the working rating tag's value without any effect. Where does QL store its ratings exactly?

In the file under `rating:email` (if configured), but note it _won't_ save immediately until forced (see #1852 for more discussion). Hence using the plugin: but you'll need to right click on the songs and select the plugin to flush tags to disk for those songs..

**Comment by sebschlicht** [comment]
> I think the key here is that the ALL-CAPS isn't working, no?

I thought the same, that's why I tried editing the rating tag `rating:quodlibet@lists.sacredchao.net` - that appeared to be working - manually, i.e. with `vorbiscomment`.

My last question was phrased wrongly. I actually meant: When and from where does QL read the ratings of a file?
That's why I was manually changing the apparently working tag (i.e. `rating:quodlibet@lists.sacredchao.net`).

Updating the file's tag did not update the rating, with or without re-scanning the library.
Removing the song and re-scanning the library, however, made QL load the rating from the file's tag!

It looks we're right, QL's tag handling is case-sensitive.
Vorbis comments [are case insensitive](https://en.wikipedia.org/wiki/Vorbis_comment), so this might be our bug.

**Comment by bassstorm** [comment]
> Updating the file's tag did not update the rating, with or without re-scanning the library.
> Removing the song and re-scanning the library, however, made QL load the rating from the file's tag!

That looks like to be caused by #3561 (fix isn't released yet afaik).

**Comment by sebschlicht** [comment]
Alright, so QL will fetch external rating updates in the future, nice!

I could not find an existing bug ticket for the case-sensitive tag handling, so this is a new one, right?

**Comment by bassstorm** [comment]
@sebschlicht seems you're right, I can't find another issue about that. Let's wait what dev team say, as I'm not familiar enough with code base.
